clang/lib/StaticAnalyzer/Core/CallEvent.cpp

```diff
@@ -797,6 +798,10 @@ RuntimeDefinition CXXInstanceCall::getRuntimeDefinition() const {
     return {};
   }
 
+  const MemRegion *R = getCXXThisVal().getAsRegion();
+  DynamicTypeInfo DynType = getDynamicTypeInfo(getState(), R);
```

Comment:
My understanding is that we do this lookup twice now and we only need the result of `DynType.canBeASubClass()`.

I wonder if it would make sense for ` getDeclForDynamicType()` to return a pair, the pointer to a declaration and a bool if the dynamic type could be a subclass to avoid the extra lookup. Other clients of that API might also want to know how precise is the dynamic type.

Reply:
That's true. I figured it's an acceptable tradeoff for removing some duplication. I'll have a look what I can do here.